gpustack/routes/worker/logs.py

```diff
@@ -107,8 +106,26 @@ async def container_log_generator(model_instance_name: str, options):
 
         # Handle different return types based on follow parameter
         if follow:
-            # When follow=True, logs_workload returns a generator
-            for log_line in log_stream:
+            # Offload blocking iteration to a background thread to avoid event loop blocking
+            try:
+                iterator = iter(log_stream)
+            except Exception as e:
+                logger.error(
+                    f"logs_workload did not return an iterable for {model_instance_name}: {e}"
+                )
+                return
+
+            while True:
+                try:
+                    log_line = await asyncio.to_thread(next, iterator)
```

Comment:
@thxCode  Is it feasible to provide async APIs in runtime? Ref: https://github.com/docker/docker-py/issues/2136#issuecomment-779314256